### PR TITLE
VC-2121 Update notification in admin panel and VC editor

### PR DESCRIPTION
**Pull request by seahunter**

### CODE REVIEW

**Review by konutis — APPROVED** [review]
JS looks ok

File: visualcomposer/Modules/Updates/UpdatesController.php [hunk 1]
```diff
@@ -65,4 +74,13 @@ protected function resetUpdateMessageCookies($upgraderObject, $options)
...
         }
     }
+
+    /**
+     * Checking version differences
+     * @return bool
+     */
+    protected function pluginCompareVersion($currentVersion)
+    {
+        return ((int)$currentVersion - (int)$this->requiredVersion) <= 4;
```
Comment (pavelthq) [inline visualcomposer/Modules/Updates/UpdatesController.php:84]: you should check if Current version is 4 versions behind latest available, not the hardcoded 34.0

Comment (pavelthq) [reply visualcomposer/Modules/Updates/UpdatesController.php:84]: naming of method should be proper:
comparePluginVersions

File: visualcomposer/Modules/Settings/NoticeController.php [hunk 2]
```diff
@@ -62,29 +62,12 @@ protected function createNotice(
 
                 $class = 'notice notice-' . $notice['type'];
 
-                if ($notice['dismissible']) {
-                    $dismissUrl = $urlHelper->adminAjax(
```
Comment (pavelthq) [inline visualcomposer/Modules/Settings/NoticeController.php:66]: what the reason to remove ajax route to dismiss notice?

File: visualcomposer/Helpers/Wp.php [hunk 3]
```diff
@@ -60,12 +60,22 @@ public function getUserSetting($userId, $key, $default = false)
...
+        return isset($updatePlugins[ VCV_PLUGIN_BASE_NAME ]) ? $updatePlugins[ VCV_PLUGIN_BASE_NAME ]->update->new_version : 0;
+    }
+
+    /**
+     * Checking version differences
+     * @return bool
+     */
+    public function isNeedUpdate()
+    {
+        return ((int)$this->getUpdateVersionFromWordpressOrg() - (int)VCV_VERSION) >= 4;
```
Comment (pavelthq) [inline visualcomposer/Helpers/Wp.php:79]: method name should isUpdateNeeded

File: visualcomposer/Helpers/Localizations.php [hunk 4]
```diff
@@ -1,1802 +1,1802 @@
...
-            'vcv:helpers:localizations:i18n:element:attributes:' . $element['tag'],
-            $genericAttributeKeys,
-            ['element' => $element]
-        );
-        $this->translatableElementAttributes[ $element['tag'] ] = $keys;
-
-        return $this->translatableElementAttributes[ $element['tag'] ];
-    }
-}
+<?php
```
Comment (pavelthq) [inline visualcomposer/Helpers/Localizations.php:1]: it seems file has changed their encoding, need to revert this changes and commit only one line. make sure you don't commit CRLF line endings

**Review by pavelthq — CHANGES_REQUESTED** [review]
